### ISSUE TITLE
NJS-110 when using type defined as TABLE OF VARCHAR2(xxx CHAR) in thin mode

### ISSUE THREAD

**Issue by Xamo1297** [oracle/node-oracledb#1646]
1. What versions are you using?

Oracle Database 19.3
    process.platform : win32
    process.version: v20.10.0
    process.arch x64
    require('oracledb').versionString: 6.3.0
    require('oracledb').oracleClientVersionString: 19.3.0.0.0 (in thick mode)

2. Is it an error or a hang or a crash?
Error

3. What error(s) or behavior you are seeing?
NJS-110: invalid Oracle type number 1 [csfrm: 129]

4. Include a runnable Node.js script that shows the problem.

SQL:
```
CREATE OR REPLACE TYPE "TABLE_OF_VARCHAR2_CHAR" AS TABLE OF VARCHAR2(100 CHAR);
CREATE OR REPLACE TYPE "TABLE_OF_VARCHAR2_BYTE" AS TABLE OF VARCHAR2(100 BYTE);
```
Javascript:
```
const oracledb = require('oracledb');

async function test() {
    try {
        const connection = await oracledb.getConnection({
            user          : "user",
            password      : "pwd",
            connectString : "server/service"
        });

        try {
            try {
                await connection.getDbObjectClass("TABLE_OF_VARCHAR2_CHAR");
                console.log("Success with TABLE_OF_VARCHAR2_CHAR");
            } catch (error) {
                console.error(`Error "${error}" for TABLE_OF_VARCHAR2_CHAR`);
            }
            try {
                await connection.getDbObjectClass("TABLE_OF_VARCHAR2_BYTE");
                console.log("Success with TABLE_OF_VARCHAR2_BYTE");
            } catch (error) {
                console.error(`Error "${error}" for TABLE_OF_VARCHAR2_BYTE`);
            }
        } finally {
            await connection.close();
        }
    } catch (error) {
        console.error(`Error ${error} connecting`);
    }
}

test();
```

Result using thin mode:
```
Error "Error: NJS-110: invalid Oracle type number 1 [csfrm: 129]" for TABLE_OF_VARCHAR2_CHAR
Success with TABLE_OF_VARCHAR2_BYTE
```

Result using thick mode:
```
Success with TABLE_OF_VARCHAR2_CHAR
Success with TABLE_OF_VARCHAR2_BYTE
```

**Comment by sudarshan12s** [comment]
Thanks @Xamo1297 for informing. This will be fixed in the upcoming 6.4.0 release.

**Comment by sharadraju** [comment]
@Xamo1297 This has been fixed in node-oracledb 6.4. Please confirm.
